### PR TITLE
Add github action to update requirements

### PR DESCRIPTION
**Pull request by madelinekosse**
It's becoming cumbersome to manage the requirements.txt file for CKAN, as we need to manually update the commit hash of our plugins. This adds a github action to create the requirements file from a template after every merge using the merge commit. Note that, as a result of this, future changes to the requirements file must be made in the template.

### CODE REVIEW

File: .github/workflows/update-requirements.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name: Update requirements.txt
+run-name: Updating commit hash for custom plugins in requirements file
+on:
+  push:
+    branches:
+      - feature/develop
```
Comment (callum-oakley) [inline .github/workflows/update-requirements.yaml:6]: so this action runs on commit to feature/develop... and itself commits to feature/develop. is it going to get stuck in a loop or is github actions smarter than that? 😆

**Review by callum-oakley — APPROVED** [review]
This looks good! ... assuming it doesn't get stuck in a loop (surely github would have thought of that? running an action on commit that makes a commit must be a common use case...)

File: .github/workflows/update-requirements.yaml (hunk 1 above)
Comment (madelinekosse) [reply .github/workflows/update-requirements.yaml:6]: Nope, I tried it on another branch and it was fine. You have to specifically tell it to trigger things with its own pushes.

File: actions/make_requirements_file.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+if [[ -z "$COMMIT" ]]; then
+    echo "Must provide COMMIT in environment" 1>&2
+    exit 1
+fi
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd)
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )";
+PARENT_DIR="${SCRIPT_DIR%/*}"
+
+sed s/commit-hash/$COMMIT/g $SCRIPT_DIR/requirements_template.txt > $PARENT_DIR/ckan/requirements.txt
```
[the remark this answers is not in the document]
Comment (madelinekosse) [reply actions/make_requirements_file.sh:9]: Am I allowed to put comments in that requirements file? Not sure if CKAN is going to like that

[the remark this answers is not in the document]
Comment (kjoshi) [reply actions/make_requirements_file.sh:9]: Looks like pip is fine with requirements.txt having comments:
https://pip.pypa.io/en/latest/reference/requirements-file-format/#comments

**Pull request merged**